pkg/retester/s3BackOffCache.go

```diff
@@ -0,0 +1,108 @@
+package retester
+
+import (
+	"bytes"
+	"fmt"
+	"github.com/aws/aws-sdk-go/aws"
+	"github.com/aws/aws-sdk-go/aws/session"
+	"github.com/aws/aws-sdk-go/service/s3"
+	"github.com/sirupsen/logrus"
+	"gopkg.in/yaml.v2"
+	"io"
+	"k8s.io/test-infra/prow/tide"
+	"time"
+)
+
+const (
+	retesterBucket = "prow-retester"
+)
+
+type s3BackOffCache struct {
+	cache          map[string]*pullRequest
+	file           string
+	cacheRecordAge time.Duration
+	logger         *logrus.Entry
+
+	awsSession *session.Session
+}
+
+func (b *s3BackOffCache) load() error {
+	return b.loadFromAwsNow(time.Now())
+}
+
+// loadFromAwsNow gets the backoff cache file from AWS S3 bucket and marshals its content into the s3BackOffCache
+func (b *s3BackOffCache) loadFromAwsNow(now time.Time) error {
+	if b.file == "" {
+		return nil
+	}
+
+	sess, err := session.NewSession(&aws.Config{
+		Region: aws.String("us-east-1"),
+	})
+	if err != nil {
+		return fmt.Errorf("couldn't create new aws session")
+	}
+	b.awsSession = sess
+
+	_, err = sess.Config.Credentials.Get()
+	if err != nil {
+		return fmt.Errorf("credentials for aws not found")
+	}
+
+	svc := s3.New(b.awsSession)
```

Comment:
Could we pass `svc` with the constructor and init `session` in the `main` function?

Reply:
We can init `session` in `retester/main.go` main function yes. And do I understand correctly you mean I should add `svc` as a field to the `s3BackOffCache struct`, so we can reuse that `svc` for each s3 operation (for loading in `loadFromAwsNow` func and also for uploading in `save` func)?

Reply:
Yes. Your understanding is correct.